event-input.c

```diff
@@ -2309,6 +2310,13 @@ evin_iomon_keypress_cb(mce_io_mon_t *iomon, gpointer data, gsize bytes_read)
             if( ev->code == KEY_POWER )
                 mce_log(LL_DEBUG, "esc key -> power key %s",
                         key_esc_down ? "press" : "release");
+        } else if( ev->code == KEY_WAKEUP ) {
+            mce_log(LL_DEVEL, "[wakeup] as gesture event");
+
+            ev->type  = EV_MSC;
+            ev->code  = MSC_GESTURE;
+            ev->value = GESTURE_DOUBLETAP;
+            datapipe_exec_full(&keypress_event_pipe, &ev);
```

Comment:
Untested as I do not have the device, but hopefully it would be enough just to block suspend instead of interfering with user activity etc. Something like:

    mce_log(LL_DEVEL, "[wakeup] block suspend a while");
    mce_wakelock_obtain(WAKEUP_EVENT_WAKELOCK_NAME,
                        WAKEUP_EVENT_TIMEOUT_MS);

With defines in constants section, something like:

    /** How long to block suspend after receiwing KEY_WAKEUP event */
    #define WAKEUP_EVENT_TIMEOUT_MS    5000
    /** Wakelock used for blocking suspend after receiving KEY_WAKEUP event */
    #define WAKEUP_EVENT_WAKELOCK_NAME "wakeup-event"